segmentation_labeling_app/transforms/transformations.py

```diff
@@ -0,0 +1,212 @@
+
+    Returns:
+        new_video: A new video down-sampled to desired FPS with selected
+        strategy
+    """
+    if os.path.exists(video_path):
+        path_parts = str(video_path).split('.')
+        extension = path_parts[len(path_parts) - 1]
+        if extension == 'h5':
+            with h5py.File(video_path, 'r') as open_video:
+                video = open_video['data'][:]
+                return _down_sample_array(full_array=video,
+                                          input_fps=input_fps,
+                                          output_fps=output_fps,
+                                          strategy=strategy)
+        else:
+            raise AttributeError('Video path provided not an h5 file, please'
+                                 'provide an h5 file')
+    else:
+        raise FileNotFoundError('Path specified doesnt exist or is not accessible')
+
+
+def _down_sample_array(full_array: np.ndarray, input_fps: int,
+                       output_fps: int, strategy: str = 'random') -> np.ndarray:
+    """
+    Function to down-sample a numpy array to a desired FPS. Follows the
+    strategy specified in the function call, current strategies are (random,
+    maximum, average). Random takes a random sample of one frame from window,
+    maximum takes the maximum frame in window, average takes the mean of window.
+    Args:
+        full_array: The video array in numpy format
+        input_fps: The FPS or Hz of the input video
+        output_fps: The desired output FPS or Hz
+        strategy: The down-sampling startegy to follow
+
+    Returns:
+        new_video: A new video down-sampled to desired FPS with selected
+        strategy
+    """
+    if output_fps > input_fps:
+        raise ValueError('Output FPS cannot be greater than input FPS')
+    sampling_ratio = int(input_fps / output_fps)
+    new_frames = []
+    for i in range(0, len(full_array), sampling_ratio):
+        frames = full_array[i:(i+sampling_ratio)]
+        if strategy == 'random':
+            new_frames.append(frames[random.randint(0, len(frames) - 1)])
+        elif strategy == 'maximum':
+            new_frames.append(np.max(frames, axis=0))
+        elif strategy == 'average':
+            new_frames.append(np.mean(frames))
+    return np.stack(new_frames)
+
+
+def get_transformed_center(coordinate_pair: Tuple[int, int],
+                           box_size: Tuple[int, int],
+                           video_shape: Tuple[int, int]):
+    """
+    Returns a coordinate pair that has been transformed to fit as closely to
+    original position as possible but still with box contained totally in video.
+    This is made to shift the coordinate if the subset box goes out of bound of
+    the video shape.
+    Args:
+        coordinate_pair: the original coordinate location for centering
+        box_size: the size of the box for subset
+        video_shape: the shape of the original video
```

Comment:
Clarify here that this is the x-y dimensions, not the true `np.shape` of the video (which would include the time axis as 0 in our data)

Reply:
good call